src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,97 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        elementData[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        size++;
+        checkIndexInBounds(index);
+        System.arraycopy(elementData, index, elementData, index + 1, size - index - 1);
+        elementData[index] = value;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            T elem = (T) list.get(i);
+            add(elem);
```

Comment:
```suggestion
            add(list.get(i));
```
Instead of creating a redundant local variable elem, you can directly pass list.get(i) as an argument to the add() method. This avoids unnecessary variable creation and improves code readability